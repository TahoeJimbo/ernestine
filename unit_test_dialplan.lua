
function runner(test, exclude_ext)
   exclude_ext = exclude_ext or ""

   print("Testing <"..test..">, excluding <"..exclude_ext..">")

   local extObj = {}
   extObj.id = "1234"
   extObj.dialstring = test
   extObj.domain = "1.2.3.4"

   results = dialplan.parse(extObj, exclude_ext)
   if (results == nil) then
      print "Results: Nil"
   else
      table_dump("Results:", results)
   end
   print ""
end

function dialplan_unittest()

   local debugState = DEBUG or nil

   DEBUG = nil


   local default_domain = nil
   local test

   test = "wait=30"
   runner(test, nil, default_domain)

   test = "8000"
   runner(test, nil, default_domain)
   runner(test, "8000", default_domain)

   test = "wait=29,8000:wait=31,4000"
   runner(test, nil, default_domain)
   runner(test, "4000", default_domain)
   runner(test, "8000", default_domain)

   test = "wait=30|8001:4001|VM(546)"
   runner(test, nil, default_domain)

   test = "wait=39|X(28)"
   runner(test, nil, default_domain)

   default_domain = "1.2.3.4"

   test = "wait=30"
   runner(test, nil, default_domain)

   test = "8000@5.6.7.8"
   runner(test, nil, default_domain)
   runner(test, "8000", default_domain)

   test = "wait=29,8000:wait=31,4000"
   runner(test, nil, default_domain)
   runner(test, "4000", default_domain)
   runner(test, "8000", default_domain)

   test = "wait=30|8001@5.6.7.8:4001%9.8.7.6|VM(546)"
   runner(test, nil, default_domain)

   test = "wait=39|X(28)"
   runner(test, nil, default_domain)



   DEBUG = debugState
end

