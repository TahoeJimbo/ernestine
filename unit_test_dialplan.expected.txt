Testing <wait=30>, excluding <>
Results:

Testing <8000>, excluding <>
Results:
1 is a table:
   dialstring	[hangup_after_bridge=true]User/8000@1.2.3.4
   kind	DS

Testing <8000>, excluding <8000>
crit: utilities_dialplan.lua: parse(): Returning nil.
Results: Nil

Testing <wait=29,8000:wait=31,4000>, excluding <>
Results:
1 is a table:
   dialstring	[call_timeout=29,hangup_after_bridge=true]User/8000@1.2.3.4:_:[call_timeout=31,hangup_after_bridge=true]User/4000@1.2.3.4
   kind	DS

Testing <wait=29,8000:wait=31,4000>, excluding <4000>
Results:
1 is a table:
   dialstring	[call_timeout=29,hangup_after_bridge=true]User/8000@1.2.3.4
   kind	DS

Testing <wait=29,8000:wait=31,4000>, excluding <8000>
Results:
1 is a table:
   dialstring	[call_timeout=31,hangup_after_bridge=true]User/4000@1.2.3.4
   kind	DS

Testing <wait=30|8001:4001|VM(546)>, excluding <>
Results:
1 is a table:
   dialstring	[call_timeout=30,hangup_after_bridge=true]User/8001@1.2.3.4:_:[call_timeout=30,hangup_after_bridge=true]User/4001@1.2.3.4
   kind	DS
2 is a table:
   dialstring	546
   kind	FU_VM

Testing <wait=39|X(28)>, excluding <>
crit: utilities_dialplan.lua: parse(): Unknown function: X
Results:

Testing <wait=30>, excluding <>
Results:

Testing <8000@5.6.7.8>, excluding <>
Results:
1 is a table:
   dialstring	[hangup_after_bridge=true]User/8000@5.6.7.8
   kind	DS

Testing <8000@5.6.7.8>, excluding <8000>
crit: utilities_dialplan.lua: parse(): Returning nil.
Results: Nil

Testing <wait=29,8000:wait=31,4000>, excluding <>
Results:
1 is a table:
   dialstring	[call_timeout=29,hangup_after_bridge=true]User/8000@1.2.3.4:_:[call_timeout=31,hangup_after_bridge=true]User/4000@1.2.3.4
   kind	DS

Testing <wait=29,8000:wait=31,4000>, excluding <4000>
Results:
1 is a table:
   dialstring	[call_timeout=29,hangup_after_bridge=true]User/8000@1.2.3.4
   kind	DS

Testing <wait=29,8000:wait=31,4000>, excluding <8000>
Results:
1 is a table:
   dialstring	[call_timeout=31,hangup_after_bridge=true]User/4000@1.2.3.4
   kind	DS

Testing <wait=30|8001@5.6.7.8:4001%9.8.7.6|VM(546)>, excluding <>
Results:
1 is a table:
   dialstring	[call_timeout=30,hangup_after_bridge=true]User/8001@5.6.7.8:_:[call_timeout=30,hangup_after_bridge=true]User/4001%9.8.7.6
   kind	DS
2 is a table:
   dialstring	546
   kind	FU_VM

Testing <wait=39|X(28)>, excluding <>
crit: utilities_dialplan.lua: parse(): Unknown function: X
Results:

